thrust/thrust/pair.h

```diff
@@ -73,10 +73,49 @@ using tuple_size = ::cuda::std::tuple_size<T>;
  *          provided by <tt>pair::second_type</tt>.
  */
 template <class T, class U>
-using pair = ::cuda::std::pair<T, U>;
-
-using ::cuda::std::get;
-using ::cuda::std::make_pair;
+struct pair : public _CUDA_VSTD::pair<T, U>
+{
+  using super_t = _CUDA_VSTD::pair<T, U>;
+  using super_t::super_t;
+
+#if (defined(_CCCL_COMPILER_GCC) && __GNUC__ < 9) || (defined(_CCCL_COMPILER_CLANG) && __clang_major__ < 12)
+  // For whatever reason nvcc complains about that constructor being used before being defined in a constexpr variable
+  constexpr pair() = default;
+
+  template <class _U1          = T,
+            class _U2          = U,
+            class _Constraints = typename _CUDA_VSTD::__pair_constraints<T, U>::template __constructible<_U1, _U2>,
+            _CUDA_VSTD::__enable_if_t<_Constraints::__implicit_constructible, int> = 0>
+  _CCCL_HOST_DEVICE constexpr pair(_U1&& __u1, _U2&& __u2)
+      : super_t(_CUDA_VSTD::forward<_U1>(__u1), _CUDA_VSTD::forward<_U2>(__u2))
+  {}
+#endif // _CCCL_COMPILER_GCC < 9 || _CCCL_COMPILER_CLANG < 12
+};
+
+#if _CCCL_STD_VER >= 2017
+template <class _T1, class _T2>
+_CCCL_HOST_DEVICE pair(_T1, _T2) -> pair<_T1, _T2>;
+#endif // _CCCL_STD_VER >= 2017
```

Comment:
Question: doesn't this fail for compilers with partial C++17 support and `-std=c++1z`?
Suggestion: test for the feature instead:
```suggestion
#ifdef __cpp_deduction_guides
template <class _T1, class _T2>
_CCCL_HOST_DEVICE pair(_T1, _T2) -> pair<_T1, _T2>;
#endif // __cpp_deduction_guides
```